crates/arroyo-connectors/src/iggy/mod.rs

```diff
@@ -0,0 +1,624 @@
+use anyhow::{anyhow, bail, Result};
+use arroyo_formats::ser::ArrowSerializer;
+use arroyo_operator::connector::Connection as ArroyoConnection;
+use arroyo_operator::operator::ConstructedOperator;
+use arroyo_rpc::api_types::connections::{
+    ConnectionProfile, ConnectionSchema, ConnectionType, TestSourceMessage,
+};
+use arroyo_rpc::var_str::VarStr;
+use arroyo_rpc::OperatorConfig;
+use iggy::client::{Client, MessageClient, StreamClient, TopicClient, UserClient};
+use iggy::clients::client::IggyClient;
+use iggy::consumer::Consumer;
+use iggy::identifier::Identifier;
+use iggy::messages::poll_messages::PollingStrategy;
+use serde::{Deserialize, Serialize};
+use std::str::FromStr;
+use std::time::{Duration, Instant};
+use tokio::sync::mpsc::Sender;
+use tracing::{info, warn};
+
+use crate::send;
+
+mod sink;
+mod source;
+
+use sink::IggySinkFunc;
+use source::IggySourceFunc;
+
+const CONFIG_SCHEMA: &str = include_str!("./profile.json");
+const TABLE_SCHEMA: &str = include_str!("./table.json");
+const ICON: &str = include_str!("./iggy.svg");
+
+typify::import_types!(
+    schema = "src/iggy/profile.json",
+    convert = {
+        {type = "string", format = "var-str"} = VarStr
+    }
+);
+
+typify::import_types!(schema = "src/iggy/table.json");
+
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub enum PartitioningStrategy {
+    PartitionId(u32),
+    Balanced,
+}
+
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub enum IggySourceOffset {
+    Earliest,
+    Latest,
+}
+
+pub struct IggyConnector {}
+
+impl arroyo_operator::connector::Connector for IggyConnector {
+    type ProfileT = IggyConfig;
+    type TableT = IggyTable;
+
+    fn name(&self) -> &'static str {
+        "iggy"
+    }
+
+    fn metadata(&self) -> arroyo_rpc::api_types::connections::Connector {
+        arroyo_rpc::api_types::connections::Connector {
+            id: "iggy".to_string(),
+            name: "Apache Iggy".to_string(),
```

Comment:
I'd prefer just "Iggy" to "Apache Iggy" as otherwise all of the apache projects would clump at the beginning of the list and be harder to find.